### PR TITLE
Fix/dial error hiding

### PR DESCRIPTION
**Pull request by whyrusleeping**
it is fairly difficult to figure out what is happening when a dial fails. This goes a little ways towards making our dialing errors clearer.

### CODE REVIEW

File: p2p/net/swarm/swarm_dial.go [hunk 1]
```diff
@@ -416,7 +417,7 @@ func (s *Swarm) dialAddrs(ctx context.Context, d *conn.Dialer, p peer.ID, remote
 	for i := 0; i < len(remoteAddrs); i++ {
 		select {
 		case exitErr = <-errs: //
-			log.Debug(exitErr)
```
Comment (jbenet) [inline p2p/net/swarm/swarm_dial.go]: :-1:  this happens all the time-- it will be super annoying for users. use notice and grep?

Comment (whyrusleeping) [inline p2p/net/swarm/swarm_dial.go]: fine... I wont do this one

**Comment by whyrusleeping** [comment]
@jbenet moved the warning back down to debug

**Comment by jbenet** [comment]
@whyrusleeping the changes to reuseport will be wiped out next time anyone updates it. ok with that? if not, we've to change it over there.

**Comment by whyrusleeping** [comment]
I'm okay with that. I really just wanted these errors to show through for a bit until we get some root cause on dial issues tracked down.

**Pull request merged**